### PR TITLE
vswitch bindings dynamic list widget

### LINKED ISSUE #35: Absolute Workspace Positions Bindings Don't Appear
**Describe the bug**
WCM doesn't show `dynamic-lists` bindings as defined by `wayfire/vswitcher`'s metadata xml file.

**Symptoms**
- Starting WCM does not seem to generate default or placeholder bindings for abs workspace pos in the `wayfire.ini` config file.
- Opening WCM doesn't show the abs workspace pos `dynamic-list`s

**Expected behavior**
- WCM should create abs workspace pos bindings in the ini file.
- WCM should show the `dynamic-lists` entries with these bindings when opened.

**wcm version**
git

### PR DESCRIPTION
**Pull request by NamorNiradnug**
Fixes #35

### CODE REVIEW

File: src/wcm.cpp [hunk 1]
```diff
@@ -648,6 +652,47 @@ BindingsDynamicList::BindingWidget::BindingWidget(const std::string & cmd_name,
...
+    Option *key_option = option->create_child_option(OPTION_PREFIX + std::to_string(
+        workspace_index), OPTION_TYPE_KEY);
+    binding_wf_opt = section->get_option(key_option->name);
+    key_entry.set_value(binding_wf_opt->get_value_str());
+    key_entry.signal_changed().connect([=]
+    {
+        key_option->set_save(key_entry.get_value());
+    });
+    label.set_alignment(Gtk::ALIGN_START);
+    // label.set_size_request(OPTION_LABEL_SIZE);
```
Comment (soreau) [inline src/wcm.cpp:668]: Can you remove this commented line of code if it is unneeded?

File: src/wcm.cpp [hunk 2]
```diff
@@ -1,21 +1,25 @@
 #include "wcm.hpp"
+#include "metadata.hpp"
```
Comment (soreau) [inline src/wcm.cpp:2]: Here this include is not needed for compilation.

File: src/wcm.cpp [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
 #include "utils.hpp"
 
 #include <libevdev/libevdev.h>
 #include <wayfire/config/compound-option.hpp>
+#include <wayfire/config/section.hpp>
```
Comment (soreau) [inline src/wcm.cpp:7]: This include seems unnecessary as well.

File: src/wcm.cpp [hunk 4]
```diff
@@ -219,7 +223,7 @@ KeyEntry::KeyEntry()
 std::ostream& operator <<(std::ostream & out, const wf::color_t & color)
 {
     return out << "rgba(" << color.r << ", " << color.g << ", " << color.b << ", " <<
-        color.a << ")";
+           color.a << ")";
```
Comment (soreau) [inline src/wcm.cpp:226]: Why is this indented? Have you rerun uncrustify on the code?

File: src/wcm.hpp [hunk 5]
```diff
@@ -27,13 +27,14 @@
 #pragma once
 
 #include <algorithm>
+#include <cstring>
```
Comment (soreau) [inline src/wcm.hpp:30]: Here it builds without this include..

File: src/wcm.hpp [hunk 6]
```diff
@@ -27,13 +27,14 @@
...
 #include <gdk/gdkwayland.h>
 #include <gtkmm.h>
 #include <iostream>
-#include <string.h>
 #include <variant>
 #include <vector>
 #include <wayfire/config/file.hpp>
+#include <wayfire/config/option.hpp>
```
Comment (soreau) [inline src/wcm.hpp:37]: and without this one. Are we using anything from these headers?

**Review by soreau — CHANGES_REQUESTED** [review]
Looks good to me, aside from the few comments.

**Comment by NamorNiradnug** [comment]
It was clangd automatically inserting needed headers.
Seems like it was a bad idea to enable this feature.

**Review by soreau — APPROVED** [review]

**Comment by soreau** [comment]
> It was clangd automatically inserting needed headers. Seems like it was a bad idea to enable this feature.

Ah, well it looks better now. Thanks!

**Pull request merged**